src/runtime/type.ts

```diff
@@ -1284,15 +1284,14 @@ export class TypeVariableInfo {
   }
 
   isAtLeastAsSpecificAs(other: TypeVariableInfo): boolean {
-    // TODO(mmandlis): add tests for this method!!!
     const thisCanReadSubset = this._canReadSubset || this._originalCanReadSubset;
     const thisCanWriteSuperset = this._canWriteSuperset || this._originalCanWriteSuperset;
     const otherCanReadSubset = other._canReadSubset || other._originalCanReadSubset;
     const otherCanWriteSuperset = other._canWriteSuperset || other._originalCanWriteSuperset;
     return ((!otherCanWriteSuperset && !thisCanWriteSuperset) ||
-            (otherCanWriteSuperset && (!thisCanWriteSuperset || otherCanWriteSuperset.isAtLeastAsSpecificAs(thisCanWriteSuperset)))) &&
+            !!(otherCanWriteSuperset && (!thisCanWriteSuperset || otherCanWriteSuperset.isAtLeastAsSpecificAs(thisCanWriteSuperset)))) &&
           ((!thisCanReadSubset && !otherCanReadSubset) ||
-            !thisCanReadSubset || thisCanReadSubset.isAtLeastAsSpecificAs(otherCanReadSubset));
+            !thisCanReadSubset || (!otherCanReadSubset || thisCanReadSubset.isAtLeastAsSpecificAs(otherCanReadSubset)));
```

Comment:
should this be (thisCanReadSubset && (!otherCanReadSubset || otherCanReadSubset.isAtLeastAsSpecificAs(otherCanReadSubset))) to match above?